src/box/watcher.c

```diff
@@ -350,7 +350,7 @@ static void
 watchable_destroy(struct watchable *watchable)
 {
 	if (watchable->worker != NULL) {
-		fiber_cancel(watchable->worker);
+		fiber_cancel_system(watchable->worker);
```

Comment:
Shouldn't system fiber use another mechanism of stopping its execution, not fiber cancel flag?

Reply:
> Shouldn't system fiber use another mechanism of stopping its execution, not fiber cancel flag?

Why do we want that? There's no difference in system fibers and usual ones (not including the flag). Could you please explain, why do we need another mechanism and what will be the difference. Setting `FIBER_IS_CANCELLED` seems to be pretty ok to me.

Reply:
It is a weak ground, but my thoughts are the following. If the fiber is marked as a system one, it means that cancelling it may lead to bad effects. I see several possible reason for that:

1. The fiber function doesn't expect cancellation or at least doesn't expect it in all yield points.
2. More actions need to be done together with cancelling the fiber to achieve a correct result.
3. The fiber must never be cancelled: neither by internal, nor by external code.

If our case is 1), I would use some more explicit mechanism (set our own flag and read it at certain points in the fiber). Otherwise it is hard to predict where the fiber will get the cancel event.

Case 2) may use 'cancel system fiber' call, but maybe an own flag would make the list of actions to stop fiber's task more explicit.

Case 3) is irrelevant for this discussion, but it is another kind of a system fiber.

So the main question here is whether we have system fibers of the type 1).